tests/checkpointing/test_legacy_checkpoints.py

```diff
@@ -14,84 +14,117 @@
+    "1.3.1",
+    "1.3.2",
+    "1.3.3",
+    "1.3.4",
+    "1.3.5",
+    "1.3.6",
+    "1.3.7",
+    "1.3.8",
+)
 
 
-# todo: add more legacy checkpoints - for < v0.8
-@pytest.mark.parametrize(
-    "pl_version",
-    [
-        # "0.8.1",
-        "0.8.3",
-        "0.8.4",
-        # "0.8.5", # this version has problem with loading on PT<=1.4 as it seems to be archive
-        # "0.9.0", # this version has problem with loading on PT<=1.4 as it seems to be archive
-        "0.10.0",
-        "1.0.0",
-        "1.0.1",
-        "1.0.2",
-        "1.0.3",
-        "1.0.4",
-        "1.0.5",
-        "1.0.6",
-        "1.0.7",
-        "1.0.8",
-        "1.1.0",
-        "1.1.1",
-        "1.1.2",
-        "1.1.3",
-        "1.1.4",
-        "1.1.5",
-        "1.1.6",
-        "1.1.7",
-        "1.1.8",
-        "1.2.0",
-        "1.2.1",
-        "1.2.2",
-        "1.2.3",
-        "1.2.4",
-        "1.2.5",
-        "1.2.6",
-        "1.2.7",
-        "1.2.8",
-        "1.2.10",
-        "1.3.0",
-        "1.3.1",
-        "1.3.2",
-        "1.3.3",
-        "1.3.4",
-        "1.3.5",
-        "1.3.6",
-        "1.3.7",
-        "1.3.8",
-    ]
-)
-def test_resume_legacy_checkpoints(tmpdir, pl_version: str):
-    path_dir = os.path.join(LEGACY_CHECKPOINTS_PATH, pl_version)
+@pytest.mark.parametrize("pl_version", LEGACY_BACK_COMPATIBLE_PL_VERSIONS)
+def test_load_legacy_checkpoints(tmpdir, pl_version: str):
+    PATH_LEGACY = os.path.join(LEGACY_CHECKPOINTS_PATH, pl_version)
+    with patch('sys.path', [PATH_LEGACY] + sys.path):
+        from simple_classif_training import ClassifDataModule, ClassificationModel
+
+        path_ckpts = sorted(glob.glob(os.path.join(PATH_LEGACY, f'*{CHECKPOINT_EXTENSION}')))
+        assert path_ckpts, 'No checkpoints found in folder "%s"' % PATH_LEGACY
+        path_ckpt = path_ckpts[-1]
+
+        model = ClassificationModel.load_from_checkpoint(path_ckpt)
+        trainer = Trainer(default_root_dir=str(tmpdir))
+        dm = ClassifDataModule()
+        res = trainer.test(model, datamodule=dm)
+        assert res[0]['test_loss'] <= 0.7
+        assert res[0]['test_acc'] >= 0.85
+        print(res)
 
-    # todo: make this as mock, so it is cleaner...
-    orig_sys_paths = list(sys.path)
-    sys.path.insert(0, path_dir)
-    from zero_training import DummyModel
 
-    path_ckpts = sorted(glob.glob(os.path.join(path_dir, f'*{CHECKPOINT_EXTENSION}')))
-    assert path_ckpts, 'No checkpoints found in folder "%s"' % path_dir
-    path_ckpt = path_ckpts[-1]
+class LimitNbEpochs(Callback):
 
-    model = DummyModel.load_from_checkpoint(path_ckpt)
-    trainer = Trainer(default_root_dir=tmpdir, max_epochs=6)
-    trainer.fit(model)
+    def __init__(self, nb: int):
+        self.limit = nb
+        self._count = 0
+
+    def on_epoch_start(self, trainer: 'pl.Trainer', pl_module: 'pl.LightningModule') -> None:
+        self._count += 1
+        if self._count >= self.limit:
+            trainer.should_stop = True
```

Comment:
Are you sure this still works on the oldest versions?

Reply:
Seems so, tested it on master